### DISCUSSION TITLE
Weird cursor behaviour

### DISCUSSION THREAD

**Discussion by asm0dey** [ghostty-org/ghostty#3855] [Feature Requests, Ideas] [answered]
In my Linux, I observe some weirdness with window decorations and the window's cursor.
On the left part of the screenshot, you can see the mouse pointer and window decoration of WezTerm; on the right, you can see the ghostty. The cursor is humongous and uncomfortable, and the decoration is okay-ish but could be smaller.
![image](https://github.com/user-attachments/assets/0c884842-b693-42cc-b6ef-8f1d1eeb55cc)
![image](https://github.com/user-attachments/assets/3a3fc3cf-1b92-4efb-8d67-35c6e1e9be43)

It's the same monitor, I have 4k 16:9 resolution with 200% scaling turned on.
CachyOS, x86_64, KDE, Wayland

Any ideas on how to fix it? Or is it a bug?

Just in case, ghostty is the only application where I observe this behaviour. I think that it has something to do with hidpi, but not sure.

**Comment by Integralist** [comment]
Not sure if this helps but you can adjust the cursor thickness and height: https://ghostty.org/docs/config/reference#adjust-cursor-thickness

**Reply by asm0dey** [reply]
Nop, doesn't influence at all. I think it is about the terminal builtin cursor, not the system one

**Comment by 00-kat** [comment]
Oh dear, this again.  A lot of other people have experienced this on the Ghostty Discord (join at https://discord.gg/ghostty): [Wayland cursor scaling issues](https://discord.com/channels/1005603569187160125/1312128716410982451), [Large cursor and visual artifacts under wayland, on arch KDE](https://discord.com/channels/1005603569187160125/1308878017409581086), [Mouse pointer bigger inside ghostty window](https://discord.com/channels/1005603569187160125/1322153965647040632), [Fractional Scaling on Linux](https://discord.com/channels/1005603569187160125/1322151734747070475)

Check if Ghostty is running under XWayland (`xlsclients`, or run `xeyes` and see if the eyes move when you move your cursor over Ghostty).  If so: XWayland apps can't scale, which is why it's like this.  Plasma has a setting somewhere which lets XWayland apps scale themselves, which would fix it.  However, I'm not sure *why* Ghostty uses XWayland in the first place for some people.

One of them solved it by reapplying their cursor.

TL;DR: no one seems to know why.


As for the decoration, [disabling libadwaita](https://ghostty.org/docs/config/reference#gtk-adwaita) is an option, and so is messing with the GTK CSS.  If in the Ghostty Discord, check out [this thread](https://discord.com/channels/1005603569187160125/1322762820441604166).  If not, try plop the following in `~/.config/gtk-4.0/gtk.css`:
> ```css
> tabbar tab,button,image,widget,label,indicator,separator,tabbox {
>     min-height:10px;
> }
> ```
(Of course, you can change `10px` to whatever CSS value you like.)

**Reply by asm0dey** [reply]
Doesn't seem so:
```
❯ xlsclients
finkel-dell  steamwebhelper
finkel-dell  steam
```

**Reply by asm0dey** [reply]
Regarding adwaita, here's my config:
```
font-family = JetBrainsMono Nerd Font
theme = catppuccin-mocha
background-opacity = 0.9
background-blur-radius = 20
gtk-adwaita = false

```
Is it correct? Do I have it disabled?

**Reply by 00-kat** [reply]
> Is it correct? Do I have it disabled?

Yes

> Doesn't seem so:

No clue then; there was someone trying to find out why in the last thread I linked ([Fractional Scaling on Linux](https://discord.com/channels/1005603569187160125/1322151734747070475)) though.

**Reply by asm0dey** [reply]
My scaling is not Fractional, it's 200%. Also, AFAIU application should scale itself according to my settings
![image](https://github.com/user-attachments/assets/12c362ab-eafb-49af-b5a7-f906bc7d28c5)

**Reply by 00-kat** [reply]
Oh apologies, I didn't realise 200% wasn't fractional scaling too.

**Reply by asm0dey** [reply]
No worries, but I'm very interested in my next steps. I am prepared to send whatever information is needed to help debugging it

**Comment by ZerGo0** [comment, marked as answer]
Installing and using the `breezeX` cursor on arch fixed this issue for me. (credits to MBA on discord for [mentioning it](https://discord.com/channels/1005603569187160125/1308878017409581086/1310715315348570203))

**Reply by asm0dey** [reply]
And this is the solution, works for me too!
For those who're looking for an exact solution, it's `yay -S breezex-cursor-theme` and then you need to select the cursor theme in the KDE settings.

**Comment by rockorager** [comment]
Just another data point:

I had this issue on Arch with GNOME installed and 100% scaling. I had to manually install `adwaita-cursors` and then set it in GNOME Tweaks. I had the large cursor issue in Sway and in GNOME (and in every libadwaita app within both of those)...

**Reply by 00-kat** [reply]
GTK only reads the dconf entry, which GNOME Tweaks sets, which is probably why it worked.  You can also use
```sh
gsettings set org.gnome.desktop.interface cursor-theme <cursor_theme_name> 
gsettings set org.gnome.desktop.interface cursor-size <cursor_theme_size>  
```
to change it.
